pokemon-service/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/service/GymServiceTest.java

```diff
@@ -0,0 +1,376 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.service;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.builders.BadgeBuilder;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.builders.GymBuilder;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.builders.TrainerBuilder;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dao.BadgeDAO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dao.GymDAO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Badge;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Gym;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Trainer;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.ChallengeStatus;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.PokemonType;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.EntityIsUsedException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.service.config.ServiceConfiguration;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.springframework.dao.DataAccessException;
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * Tests for gym service.
+ *
+ * @author Jiří Medveď 38451
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class GymServiceTest {
+
+    @Mock
+    private GymDAO gymDAO;
+
+    @Mock
+    private BadgeDAO badgeDAO;
+
+    @Inject
+    @InjectMocks
+    private GymServiceImpl gymService;
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    //Existing gyms
+    private Gym gymPraha;
+    private Gym gymBrno;
+
+    //Gym to be created
+    private Gym gymOstrava;
+
+    //Gym to throw DataAccessException
+    private Gym errorneousGym;
+
+    //Gym Leaders
+    private Trainer gymLeaderPraha;
+    private Trainer gymLeaderBrno;
+    private Trainer gymLeaderOstrava;
+
+    //Trainer without gym
+    private Trainer otherTrainer;
+
+    private final DataAccessException dae = new DataAccessException("throw") {
```

Comment:
This exception should be emitted on service layer, for purposes of checking if it is thrown, the DAO should throw anything else, e.g. PersistenceException.